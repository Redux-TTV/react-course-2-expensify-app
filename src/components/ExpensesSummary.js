import React from 'react';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total.js';

export const ExpensesSummary = (expenses) => {
  const expensesTotal = getExpensesTotal(expenses.expenses);
  const expenseCount = expenses.expenses.length;
  return (
    <div>
      <p>
        Viewing {expenseCount} expenses totalling {numeral(expensesTotal/100).format('$0,0.00')}
      </p>
    </div>
  );
};
