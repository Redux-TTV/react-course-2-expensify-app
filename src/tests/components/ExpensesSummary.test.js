import expenses from '../fixtures/expenses.js';
import {ExpensesSummary} from '../../components/ExpensesSummary.js';
import React from 'react';
import {shallow} from 'enzyme';

test('should render ExpensesSummary component correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
})
