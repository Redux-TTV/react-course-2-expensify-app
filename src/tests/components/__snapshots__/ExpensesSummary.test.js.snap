// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component correctly 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,144.50
  </p>
</div>
`;
